src/workerd/api/worker-rpc.c++

```diff
@@ -62,50 +122,50 @@ kj::Promise<capnp::Response<rpc::JsRpcTarget::CallResults>> WorkerRpc::sendWorke
   }
 
   auto callResult = builder.send();
-  auto customEventResult = worker->customEvent(kj::mv(event)).attach(kj::mv(worker));
-
-  // If customEvent throws, we'll cancel callResult and propagate the exception. Otherwise, we'll
-  // just wait until callResult finishes.
-  return callResult.exclusiveJoin(customEventResult
-      .then([](auto&&) -> kj::Promise<capnp::Response<rpc::JsRpcTarget::CallResults>> {
-        return kj::NEVER_DONE;
-      }));
+
+  return ioContext.awaitIo(js, kj::mv(callResult),
+      [](jsg::Lock& js, auto result) -> jsg::Value {
+    return jsg::Value(js.v8Isolate, deserializeV8(js, result.getResult().getV8Serialized()));
+  });
 }
 
-kj::Maybe<WorkerRpc::RpcFunction> WorkerRpc::getRpcMethod(jsg::Lock& js, kj::StringPtr name) {
-  // Named intercept is enabled, this means we won't default to legacy behavior.
-  // The return value of the function is a promise that resolves once the remote returns the result
-  // of the RPC call.
-  return RpcFunction([this, methodName=kj::str(name)]
+kj::Maybe<JsRpcCapability::RpcFunction> JsRpcCapability::getRpcMethod(
```

Comment:
May be missing something, but is this method actually needed? AFAICT we only call `Fetcher::getRpcMethod()`, and never actually construct a `JsRpcCapability`, we just call the static `JsRpcCapability::sendJsRpc()`. It seems like all we actually need is the static method, which could just be a function in `worker-rpc.c++` instead, no?

Judging by some comments in `worker-rpc.h` I figure it may be relevant in the future when we're able to pass capabilities around, but I'm wondering if the intention is to use this class then, or just put more stuff on `Fetcher`? If we're going to use `Fetcher` then this class isn't needed anymore.

Reply:
Sorry, started replying to this earlier then got distracted.

Currently, there's no way to construct a `JsRpcCapability`, but that will change soon. This is the type we will use when people pass an object with methods as one of the parameters or results of some other RPC call. The receiving end will get a `JsRpcCapability` that makes RPCs back to the sender.